includes/components/class-attribute.php

```diff
@@ -307,6 +310,72 @@ public function register_models() {
 		}
 	}
 
+	/**
+	 * Validates unique model attributes.
+	 *
+	 * @param array  $errors Error messages.
+	 * @param object $model Model object.
+	 * @return array
+	 */
+	public function validate_unique_attributes( $errors, $model ) {
+		if ( empty( $errors ) ) {
+
+			// Set unique fields.
+			$fields = [];
+
+			foreach ( $model->_get_fields() as $field_name => $field ) {
+				if ( ! hp\get_array_value( $field->get_args(), 'unique' ) || ! $field->get_value() ) {
+					continue;
+				}
+
+				$fields[] = [
```

Comment:
You can use array_filters for _get_fields instead and use an array of field objects, then use common methods like get_name(), get_label(), get_value(), creating arrays with custom keys adds another level of abstraction, otherwise if you know that it's an array of framework-common objects you can use methods without checking the place where an array is filled